cpp/include/cudf/detail/aggregation/aggregation.hpp

```diff
@@ -355,6 +355,12 @@ struct target_type_impl<Source, aggregation::ROW_NUMBER> {
   using type = cudf::size_type;
 };
 
+// Always use list for COLLECT
+template <typename Source>
+struct target_type_impl<Source, aggregation::COLLECT> {
+  using type = cudf::list_view;
+};
+
```

Comment:
Is this actually needed? The `target_type` logic is for determining the type to use for an accumulator for ops like sum/min/max. I wouldn't think it would be needed for `COLELCT`.

Reply:
Yes -- otherwise we fail this check: https://github.com/rapidsai/cudf/blob/branch-0.16/cpp/src/groupby/groupby.cu#L102

which eventually calls: https://github.com/rapidsai/cudf/blob/5fbffb6688f8df7a395f54fa552a615a70346b12/cpp/include/cudf/detail/aggregation/aggregation.hpp#L528